library/alloc/src/collections/btree/map.rs

```diff
@@ -1361,36 +1372,48 @@ impl<K, V> IntoIterator for BTreeMap<K, V> {
     type Item = (K, V);
     type IntoIter = IntoIter<K, V>;
 
-    fn into_iter(self) -> IntoIter<K, V> {
-        let mut me = ManuallyDrop::new(self);
```

Comment:
This `ManuallyDrop<BTreeMap<..>>` *was* useful before -- since `Drop` was implemented with `into_iter`, it would be infinitely recursive to trigger a new drop from here.

With your changes it's less severe -- it will try to drop `self`, but that will see `root` as `None` and have nothing left to do. We could choose to still avoid that useless drop, but it's not a big deal.

Reply:
Okay, thanks, I didn't understand it after all.